cfme/tests/cloud_infra_common/test_tag_objects.py

```diff
@@ -1,128 +1,156 @@
 from cfme.infrastructure.provider import InfraProvider
 from cfme.infrastructure.virtual_machines import Vm, Template
-from cfme.web_ui import mixins
-from cfme.modeling.base import BaseCollection
+from fixtures.provider import setup_one_or_skip
+from cfme.utils.providers import ProviderFilter
 from cfme.utils.appliance.implementations.ui import navigate_to
-from cfme.exceptions import ItemNotFound
-
-
-param_classes = {
-    # TODO: replace other classes with collections
-    'Infra Providers': InfraProvider,
-    'Infra VMs': Vm,
-    'Infra Templates': Template,
-    'Infra Hosts': HostCollection,
-    'Infra Clusters': ClusterCollection,
-    'Infra Stores': DatastoreCollection,
-
-    'Cloud Providers': CloudProvider,
-    'Cloud Instances': Instance,
-    # 'Cloud Flavors': Flavor, # Test needs to be refactored along with tag mixin for widgets
-    # 'Cloud Availabity Zones': AvailabilityZone,  # Add back when all classes support widgets
-    'Cloud Tenants': TenantCollection
-}
-
-pytestmark = [
-    pytest.mark.parametrize("location", param_classes),
-    pytest.mark.tier(3)
-]
-
-
-def _navigate_and_check(location, appliance):
-    if issubclass(param_classes[location], BaseCollection):
-        cls_or_col = param_classes[location](appliance)
-    else:
-        cls_or_col = param_classes[location]
-    view = navigate_to(cls_or_col, 'All')
-    view.toolbar.view_selector.select('Grid View')
-    try:
-        entity = view.entities.get_first_entity()
-        entity.check()
-        return entity
-    except ItemNotFound:
-        pytest.skip("No Quadicon present, cannot test.")
 
-# TODO Replace navigation and item selection with widgets when all tested classes have them
 
+infra_test_items = [
+    ('infra_provider', InfraProvider),
+    ('vms', Vm),
+    ('templates', Template),
+    ('hosts', None),
+    ('clusters', None),
+    ('datastores', None)
+]
 
-def _tag_item_through_selecting(request, location, tag, appliance):
-    """Add a tag to an item with going through the details page.
+cloud_test_items = [
+    ('cloud_provider', CloudProvider),
+    ('instances', Instance),
+    ('flavors', Flavor),
+    ('availability_zones', AvailabilityZone),
+    ('cloud_tenants', None),
+    ('keypairs', None),
+    ('templates', Image)
+]
 
-    Prerequisities:
-        * Have a tag category and tag created.
-        * Be on the page you want to test.
 
-    Steps:
-        * Select any quadicon.
-        * Select ``Policy/Edit Tags`` and assign the tag to it.
-        * Click on the quadicon and verify the tag is assigned. (TODO)
-        * Go back to the quadicon view and select ``Policy/Edit Tags`` and remove the tag.
-        * Click on the quadicon and verify the tag is not present. (TODO)
-    """
-    _navigate_and_check(location, appliance)
+@pytest.fixture(scope='module')
+def infra_provider(request):
+    prov_filter = ProviderFilter(classes=[InfraProvider])
+    return setup_one_or_skip(request, filters=[prov_filter])
```

Comment:
**REQUIRED**
This fixture is not needed. You can use `@pytest.mark.provider` and `provider` fixture.